casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -742,17 +746,37 @@ class RuntimeManagerImpl[F[_]: Concurrent: Metrics: Span: Log](
     // Create a deploy with newly created private key
     val (privKey, _) = Secp256k1.newKeyPair
     val deploy       = ConstructDeploy.sourceDeployNow(bondsQuerySource, sec = privKey)
-    captureResults(hash, deploy)
-      .ensureOr(
-        bondsPar =>
-          new IllegalArgumentException(
-            s"Incorrect number of results from query of current bonds in state ${PrettyPrinter
-              .buildString(hash)}: ${bondsPar.size}"
-          )
-      )(bondsPar => bondsPar.size == 1)
-      .map { bondsPar =>
-        toBondSeq(bondsPar.head)
-      }
+    def logError(err: Throwable, details: RetryDetails): F[Unit] = details match {
+      case WillDelayAndRetry(_, retriesSoFar: Int, _) =>
+        Log[F].error(
+          s"Unexpected exception during computeBonds. Retrying ${retriesSoFar + 1} time."
+        )
+      case GivingUp(totalRetries: Int, _) =>
+        Log[F].error(
+          s"Unexpected exception during computeBonds. Giving up after ${totalRetries} retries."
+        )
+    }
+
+    implicit val s = new retry.Sleep[F] {
+      override def sleep(delay: FiniteDuration): F[Unit] = ().pure[F]
+    }
+
+    retryingOnAllErrors[Seq[Bond]](
+      RetryPolicies.limitRetries[F](5),
+      onError = logError
+    )(
+      captureResults(hash, deploy)
+        .ensureOr(
+          bondsPar =>
+            new IllegalArgumentException(
+              s"Incorrect number of results from query of current bonds in state ${PrettyPrinter
+                .buildString(hash)}: ${bondsPar.size}"
+            )
+        )(bondsPar => bondsPar.size == 1)
+        .map { bondsPar =>
+          toBondSeq(bondsPar.head)
+        }
+    )
```

Comment:
Please add some TODO note that this is temp debugging solution.

Reply:
https://github.com/rchain/rchain/pull/3176/commits/fc687229328b098d0394c087559d8eafda606595#diff-723ebc1ad7e9c98d566f4d96538fbb911b413c587f9b27e1ec4bc947efb34820R764